outport/process/transactionsfee/builtInCost.go

```diff
@@ -0,0 +1,89 @@
+package transactionsfee
+
+import (
+	"github.com/mitchellh/mapstructure"
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+// ArgsBuiltInFunctionCost holds all components that are needed to create a new instance of builtInFunctionsCost
+type ArgsBuiltInFunctionCost struct {
+	GasSchedule core.GasScheduleNotifier
+	ArgsParser  process.ArgumentsParser
+}
+
+type builtInFunctionsCost struct {
+	gasConfig *process.GasCost
+}
+
+// NewBuiltInFunctionsCost will create a new instance of builtInFunctionsCost
+func NewBuiltInFunctionsCost(gasSchedule core.GasScheduleNotifier) (*builtInFunctionsCost, error) {
+	if check.IfNil(gasSchedule) {
+		return nil, process.ErrNilGasSchedule
+	}
+
+	bs := &builtInFunctionsCost{}
+
+	var err error
+	bs.gasConfig, err = createGasConfig(gasSchedule.LatestGasSchedule())
+	if err != nil {
+		return nil, err
+	}
+
+	gasSchedule.RegisterNotifyHandler(bs)
+
+	return bs, nil
+}
+
+// GasScheduleChange is called when gas schedule is changed, thus all contracts must be updated
+func (bc *builtInFunctionsCost) GasScheduleChange(gasSchedule map[string]map[string]uint64) {
+	newGasConfig, err := createGasConfig(gasSchedule)
+	if err != nil {
+		return
+	}
+
+	bc.gasConfig = newGasConfig
```

Comment:
this is not concurrent safe.. gasConfig should be protected by a mutex + concurrency unittest

```go
func TestBuiltInFunctionsCost_Concurrent(t *testing.T) {
	gs := defaults.FillGasMapInternal(map[string]map[string]uint64{}, 100)
	builtInCostHandler, err := NewBuiltInFunctionsCost(testscommon.NewGasScheduleNotifierMock(gs))
	require.Nil(t, err)

	wg := sync.WaitGroup{}
	numCalls := 100
	wg.Add(numCalls)
	for i := 0; i < numCalls; i++ {
		go func(idx int) {
			switch idx % 2 {
			case 0:
				builtInCostHandler.GasScheduleChange(gs)
			case 1:
				builtInCostHandler.GetESDTTransferBuiltInCost()
			}
			wg.Done()
		}(i)
	}

	wg.Wait()
}
```

Reply:
added mutex